src/index.ts

```diff
@@ -673,6 +682,98 @@ export class ERC725 {
   encodeKeyName(keyName: string, dynamicKeyParts?: DynamicKeyParts): string {
     return encodeKeyName(keyName, dynamicKeyParts);
   }
-}
 
+  baseConfig = { ipfsGateway: IPFS_GATEWAY_BASE_URL };
+
+  lspTypeOptions: Record<Exclude<LSPType, LSPType.Unknown>, LspTypeOption> = {
+    [LSPType.LSP3UniversalProfile]: {
+      interfaceId: INTERFACE_IDS.LSP0ERC725Account,
+      lsp2Schema: this.getSupportedStandardObject(
+        lsp3Schema as ERC725JSONSchema[],
+      ),
+    },
+    [LSPType.LSP7DigitalAsset]: {
+      interfaceId: INTERFACE_IDS.LSP7DigitalAsset,
+      lsp2Schema: this.getSupportedStandardObject(
+        lsp4Schema as ERC725JSONSchema[],
+      ),
+    },
+    [LSPType.LSP8IdentifiableDigitalAsset]: {
+      interfaceId: INTERFACE_IDS.LSP8IdentifiableDigitalAsset,
+      lsp2Schema: this.getSupportedStandardObject(
+        lsp4Schema as ERC725JSONSchema[],
+      ),
+    },
+    [LSPType.LSP9Vault]: {
+      interfaceId: INTERFACE_IDS.LSP9Vault,
+      lsp2Schema: this.getSupportedStandardObject(
+        lsp9Schema as ERC725JSONSchema[],
+      ),
+    },
+  };
+
+  private getSupportedStandardObject(schemas: ERC725JSONSchema[]) {
+    try {
+      const results = schemas.filter((schema) => {
+        return schema.name.startsWith('SupportedStandard:');
+      });
+
+      if (results.length === 0) {
+        return null;
+      }
+
+      return results[0];
+    } catch (error) {
+      return null;
+    }
+  }
+
+  async detectLSPInterface(
+    contractAddress: string,
+    lspType: LSPType,
+    web3Provider: Web3,
+  ) {
+    if (lspType === LSPType.Unknown) {
+      return false;
+    }
+
+    // EIP-165 detection
+    const contract = new web3Provider.eth.Contract(
+      COMMON_ABIS.supportsInterface as AbiItem[],
+      contractAddress,
+    );
+
+    // Check if the contract implements the LSP interface ID
+    let doesSupportInterface: boolean;
```

Comment:
what about `isInterfaceSupported`?

Reply:
we can leave it out, initially i planned on having a boolean but in the latest version due to modularity, we just return plain values